sdk/python/feast/infra/online_stores/contrib/hbase_online_store/hbase.py

```diff
@@ -0,0 +1,232 @@
+        ],
+        progress: Optional[Callable[[int], Any]],
+    ) -> None:
+        """
+        Write a batch of feature rows to Hbase online store.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            table: Feast FeatureView.
+            data: a list of quadruplets containing Feature data. Each quadruplet contains an Entity Key,
+            a dict containing feature values, an event timestamp for the row, and
+            the created timestamp for the row if it exists.
+            progress: Optional function to be called once every mini-batch of rows is written to
+            the online store. Can be used to display progress.
+        """
+
+        hbase = HbaseUtils(self._get_conn(config))
+        project = config.project
+        table_name = _table_id(project, table)
+
+        b = hbase.batch(table_name)
+        for entity_key, values, timestamp, created_ts in data:
+            row_key = serialize_entity_key(entity_key).hex()
+            values_dict = {}
+            for feature_name, val in values.items():
+                values_dict[
+                    HbaseConstants.get_col_from_feature(feature_name)
+                ] = val.SerializeToString()
+            if isinstance(timestamp, datetime):
+                values_dict[HbaseConstants.DEFAULT_EVENT_TS] = struct.pack(
+                    ">L", int(calendar.timegm(timestamp.timetuple()))
+                )
+            else:
+                values_dict[HbaseConstants.DEFAULT_EVENT_TS] = timestamp
+            if created_ts is not None:
+                if isinstance(created_ts, datetime):
+                    values_dict[HbaseConstants.DEFAULT_CREATED_TS] = struct.pack(
+                        ">L", int(calendar.timegm(created_ts.timetuple()))
+                    )
+                else:
+                    values_dict[HbaseConstants.DEFAULT_CREATED_TS] = created_ts
+            b.put(row_key, values_dict)
+        b.send()
+
+    def online_read(
+        self,
+        config: RepoConfig,
+        table: FeatureView,
+        entity_keys: List[EntityKeyProto],
+        requested_features: Optional[List[str]] = None,
+    ) -> List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]]:
+        """
+        Retrieve feature values from the Hbase online store.
+
+        Args:
+            config: The RepoConfig for the current FeatureStore.
+            table: Feast FeatureView.
+            entity_keys: a list of entity keys that should be read from the FeatureStore.
+        """
+        hbase = HbaseUtils(self._get_conn(config))
+        project = config.project
+        table_name = _table_id(project, table)
+
+        result: List[Tuple[Optional[datetime], Optional[Dict[str, ValueProto]]]] = []
+
+        for entity_key in entity_keys:
+            row_key = serialize_entity_key(entity_key).hex()
+            val = hbase.row(table_name, row_key=row_key)
```

Comment:
any reason you aren't using hbase.rows?

Reply:
Miss from my side. Changed rto rows and then iterated over the rows to fetch the results. This would reduce too many calls to hbase. Good Suggestion!